src/processout/processout.ts

```diff
@@ -348,21 +348,67 @@ module ProcessOut {
          * @param  {callback} error
          * @return {void}
          */
-        public tokenize(val: Card | CardForm | ApplePay, data: any,
+        public tokenize(val: Card | CardForm | ApplePay | NetworkToken, data: any,
             success: (token: string)  => void,
             error:   (err: Exception) => void): void {
 
             if (val instanceof Card)
                 return this.tokenizeCard(<Card>val, data, success, error);
             if (val instanceof CardForm)
                 return this.tokenizeForm(<CardForm>val, data, success, error);
+            if (val instanceof NetworkToken)
+                return this.tokenizeNetworkToken(<NetworkToken>val, success, error);
             if (val instanceof ApplePay)
                 return (<ApplePay>val).tokenize(data, success, error);
 
+
             throw new Exception("processout-js.invalid-type",
                 "The first parameter had an unknown type/instance. The value must be an instance of either Card, CardForm or ApplePay.");
         }
 
+        /**
+         * TokenizeNetworkToken takes the network token payload
+         * and encodes in base64. Then it creates a ProcessOut
+         * token that can be sent to your server and used to charge your
+         * customer
+         * @param  {string} payload
+         * @param  {callback} success
+         * @param  {callback} error
+         * @return {void}
+         */
+         protected tokenizeNetworkToken(token: NetworkToken,
+            success: (token: string) => void,
+            error:   (err: Exception) => void): void {
+
+            let data = {}
+            let tokenType = token.getTokenType();
+            
+            switch(tokenType) {
+                case TokenType.GooglePay:
+                let encodedPayload = btoa(JSON.stringify(token.getPayload()));
+
+                data = {
+                    token_type: tokenType,
+                    payment_token: encodedPayload
+                };
+                break;
```

Comment:
This `case` block should have an additional level of indentation. For more information, refer to this section of the Google JavaScript Style Guide:

https://google.github.io/styleguide/jsguide.html#formatting-switch-statements